### PR TITLE
refactor tagger entity id prefixes to its own type

### PR DESCRIPTION
**Pull request by adel121**
### What does this PR do?

The tagger builds on predefined entity id prefixes to uniquely identify tag entities in the tagstore. 

The available prefixes are fixed and defined [here](https://github.com/DataDog/datadog-agent/tree/main/comp/core/tagger#entity-ids).

This PR refactors the usage of these prefixes so that it is uniform and based on a distinct type.

### Motivation

- Unify usage of entity id prefixes.
- Preparatory step for adding subscription filters to the tagger (we should be able to list possible prefixes, include and exclude them) 

### Additional Notes

There are other places in the codebase that build the entity id manually. We can refactor them to use the new `EntityIDPrefix` type in a separate PR to homogenise building the tag entity id.

### Possible Drawbacks / Trade-offs

None

### Describe how to test/QA your changes

No need. Handled by unit tests and E2E.

### CODE REVIEW

**Review by wdhif — APPROVED** [review]

File: comp/core/tagger/common/constants.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package common provides common constants and methods for the tagger component and implementation
+package common
+
+import "github.com/DataDog/datadog-agent/comp/core/tagger/types"
+
+const (
```
Comment (davidor) [inline comp/core/tagger/common/constants.go:11]: I think we should check in the code the usages of this hardcoded constants and replace them with the ones here. There are some usages of .Tag that hardcode the prefix for example.

**Review by davidor — APPROVED** [review]

**Pull request merged**